### PR TITLE
740/2507 - Add an option to clear the field

### PR DESCRIPTION
**Pull request by ericangeles**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Added an ability to clear the field in lookup component.  It supports both single and multiple values.

Make `clearable: true`, it will show `x` button in the field (It inherits in searchfield component).

Also adds `apply` button in `selectable: single`

**Related github/jira issue (required)**:
Closes
- #740
- #2507

**Steps necessary to review your pull request (required)**:
- Pull this branch, build, and run the app
- Navigate to http://localhost:4000/components/lookup/example-clearable.html

1.) Single Select

Scenario 1
- Click the first field `Products (Single)` icon to trigger the lookup modal
- Notice, it has already an `apply` button. Select any item
- That item should now populated in the field and that will show the `x` icon button
- Open again the lookup modal
- Deselect the item and hit `apply` button
- Should clear the in the field

Scenario 2
- Open and select an item
- It will show it again in the field
- Click `x` button. Should clear the field
- Try to type anything, should trigger the `x` button and can clear that too
- Test this on uplift

2.) Multiple Select
- Open the lookup modal
- Select all and hit `apply` button
- `x` button should show, text should truncated, and it should not overlap on those icons
- Hit `x` button, should clear the field
- Test this also on uplift

**Included in this Pull Request**:
- [x] A note to the change log.

### CODE REVIEW

**Comment by tmcconechy** [comment]
yes @deep7102 thats what i mean. maybe we want it since we have the button. But im not sure and this is how it was so lets leave it for now and see

File: src/components/lookup/_lookup.scss [hunk 1]
```diff
@@ -75,6 +75,28 @@
   }
 }
 
+.lookup-wrapper {
+  .searchfield-wrapper {
+    > .icon.close {
+      right: 30px;
```
Comment (EdwardCoyle) [inline src/components/lookup/_lookup.scss:81]: These new CSS rules need a corresponding rule for `html[dir='rtl']` situations:  http://localhost:4000/components/lookup/example-clearable.html?locale=ar-EG looks a bit funny.

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 ### v4.29.0 Features
 
+- `[Lookup]` Added x icon button so it has the ability to clear the field. ([#740](https://github.com/infor-design/enterprise/issues/740))
+- `[Lookuo]` Added x icon and apply button inside of modal so it has a two options to clear the field. ([#2507](https://github.com/infor-design/enterprise/issues/2507))
```
[the remark this answers is not in the document]
Comment (tmcconechy) [reply docs/CHANGELOG.md]: I had made edits here.. Just pushed

**Comment by ericangeles** [comment]
All good now. Take a peek, @tmcconechy @EdwardCoyle @deep7102. Thanks y'all! :)

**Comment by EdwardCoyle** [comment]
:banana: thanks @ericangeles

**Pull request merged**